### ISSUE TITLE
Incorrect "Syntax Error: attributes are not allowed on IF_EXPR"

### ISSUE THREAD

**Issue by d4h0** [rust-analyzer/rust-analyzer#4874]
```rust
    if original_name == ident_name.to_string() {
        None
    } else {
        Some(quote! {#[serde(rename = #original_name)]})
    }
```

Possible duplicate of https://github.com/rust-analyzer/rust-analyzer/issues/4386, but because my attribute usage is within a macro, I submit this as a new issue.

The following: 

> rustc accepts this. Also the syntax error is reported on the closing }, not the #[cfg(false_)] or if.

from the above issue is true here as well.

Is there a way to suppress this error message?

**Comment by bjorn3** [comment]
> Possible duplicate of #4386, but because my attribute usage is within a macro, I submit this as a new issue.

This is the same issue. It doesn't matter for the parser if this is part of a macro expansion or written by yourself.

> Is there a way to suppress this error message?

No, somebody will need to implement this.

**Comment by d4h0** [comment]
> This is the same issue. It doesn't matter for the parser if this is part of a macro expansion or written by yourself.

Okay, thanks. In this case, I will close this issue and add my example to the other issue.
